frontend/src/metabase/dashboard/components/PublicLinkPopover/PublicLinkPopover.unit.spec.tsx

```diff
@@ -0,0 +1,231 @@
+import userEvent from "@testing-library/user-event";
+import { useState } from "react";
+import { renderWithProviders, screen } from "__support__/ui";
+import { createMockUser } from "metabase-types/api/mocks";
+import { createMockState } from "metabase-types/store/mocks";
+import type {
+  ExportFormatType,
+  PublicLinkPopoverProps,
+} from "./PublicLinkPopover";
+import { PublicLinkPopover } from "./PublicLinkPopover";
+
+// https://github.com/nkbt/react-copy-to-clipboard/issues/106#issuecomment-605227151
+jest.mock("copy-to-clipboard", () => {
+  return jest.fn();
+});
+
+const TestComponent = ({
+  createPublicLink,
+  deletePublicLink,
+  extensions,
+  getPublicLink,
+  isOpen: mockIsOpen,
+  onClose: mockIsClosed,
+  uuid,
+}: Omit<PublicLinkPopoverProps, "target">) => {
+  const target = (
+    <button data-testid="target" onClick={() => setIsOpen(true)}>
+      Target
+    </button>
+  );
+  const [isOpen, setIsOpen] = useState(mockIsOpen);
+
+  const onClose = () => {
+    setIsOpen(false);
+    mockIsClosed();
+  };
+
+  return (
+    <PublicLinkPopover
+      target={target}
+      createPublicLink={createPublicLink}
+      deletePublicLink={deletePublicLink}
+      uuid={uuid}
+      getPublicLink={getPublicLink}
+      isOpen={isOpen}
+      onClose={onClose}
+      extensions={extensions}
+    />
+  );
+};
+
+const setup = ({
+  hasUUID = true,
+  isOpen = true,
+  extensions = [],
+  isAdmin = false,
+}: {
+  hasUUID?: boolean;
+  isOpen?: boolean;
+  extensions?: ExportFormatType[];
+  isAdmin?: boolean;
+} = {}) => {
+  const createPublicLink = jest.fn();
+  const deletePublicLink = jest.fn();
+  const onClose = jest.fn();
+  const getPublicLink = jest.fn(
+    ({ exportFormat }: { exportFormat: ExportFormatType }) =>
+      exportFormat
+        ? `sample-public-link.${exportFormat}`
+        : "sample-public-link",
+  );
+
+  renderWithProviders(
+    <TestComponent
+      createPublicLink={createPublicLink}
+      deletePublicLink={deletePublicLink}
+      uuid={hasUUID ? "mock-uuid" : null}
+      getPublicLink={getPublicLink}
+      isOpen={isOpen}
+      onClose={onClose}
+      extensions={extensions}
+    />,
+    {
+      storeInitialState: createMockState({
+        currentUser: createMockUser({
+          is_superuser: isAdmin,
+        }),
+      }),
+    },
+  );
+  return {
+    createPublicLink,
+    deletePublicLink,
+    getPublicLink,
+    onClose,
+  };
+};
+
+describe("PublicLinkPopover", () => {
+  describe("when rendering for admins vs non-admins", () => {
+    it("should render public link information for admins", async () => {
```

Comment:
non blocking suggestion:
I find test suites following the [given when then](https://martinfowler.com/bliki/GivenWhenThen.html) a bit more readable
In this example what do you think about this structure?
```
(given a) PublicLinkPopover
	when the user is an admin
	  it should render public link information
	  it should render `Remove this public link`
	when the user is not an admin
	  it should render public link information
	  it should not render `Remove this public link`
...
```

Reply:
Gotcha, I was looking at that same article yesterday and it does make sense, the semantics just looked weird to me, but I do like the structure you've laid out. I'll make sure the tests pass first before making any changes

Reply:
Yeah when I first read it it looked a bit abstract to me, now I just consider the "given x" as "given this component i need to test", and the "when" as "when we are in scenario X" and it usually helps me writing down the tests